job/steps/semantic_similarity.py

```diff
@@ -46,3 +53,13 @@ def preprocess_data(site: Site, article_data: List[Dict[str]], interactions_data
     df = df_metadata.merge(df_data, how="inner", on="external_id")
 
     return df
+
+
+def generate_embeddings(train_data: pd.DataFrame) -> np.ndarray:
+    """
+    Given article data DataFrame with a text column, create article-level text embeddings.
+    """
+    texts = train_data["text"].tolist()
+    model = SentenceTransformer(PRETRAINED_MODEL_NAME, device="cpu")
+
+    return model.encode(texts, convert_to_numpy=True)
```

Comment:
❓ I'm guessing the answer is yes, but just to ask: will we always want to `convert_to_numpy`?